test/functional/editions_controller_test.rb

```diff
@@ -167,37 +136,149 @@ class EditionsControllerTest < ActionController::TestCase
       should "redirect to edition_path when user does not have govuk-editor permission" do
         user = FactoryBot.create(:user, :welsh_editor, name: "Stub User")
         login_as(user)
-        get :confirm_unpublish, params: { id: @guide.id, redirect_url: nil }
+        get :confirm_unpublish, params: { id: @edition.id, redirect_url: nil }
 
-        assert_redirected_to edition_path(@guide)
+        assert_redirected_to edition_path(@edition)
       end
 
       should "show success message and redirect to root path when unpublished successfully with redirect url" do
-        get :process_unpublish, params: { id: @guide.id, redirect_url: "https://www.gov.uk/redirect-to-replacement-page" }
+        get :process_unpublish, params: { id: @edition.id, redirect_url: "https://www.gov.uk/redirect-to-replacement-page" }
 
         assert_equal "Content unpublished and redirected", flash[:success]
       end
 
       should "show success message and redirect to root path when unpublished successfully without redirect url" do
         UnpublishService.stubs(:call).returns(true)
-        get :process_unpublish, params: { id: @guide.id, redirect_url: nil }
+        get :process_unpublish, params: { id: @edition.id, redirect_url: nil }
 
         assert_equal "Content unpublished", flash[:success]
       end
 
       should "show error message when unpublish is unsuccessful" do
         UnpublishService.stubs(:call).returns(nil)
-        get :process_unpublish, params: { id: @guide.id, redirect_url: nil }
+        get :process_unpublish, params: { id: @edition.id, redirect_url: nil }
 
         assert_select ".gem-c-error-summary__list-item", "Due to a service problem, the edition couldn't be unpublished"
       end
 
       should "show error message when unpublish service returns an error" do
         UnpublishService.stubs(:call).raises(StandardError)
-        get :process_unpublish, params: { id: @guide.id, redirect_url: nil }
+        get :process_unpublish, params: { id: @edition.id, redirect_url: nil }
 
         assert_select ".gem-c-error-summary__list-item", "Due to a service problem, the edition couldn't be unpublished"
       end
     end
   end
+
+  context "#admin" do
+    should "show the admin page for the edition" do
+      get :admin, params: { id: @edition.id }
+
+      assert_response :success
+    end
+
+    context "Welsh editors" do
+      setup do
+        login_as_welsh_editor
+        @welsh_guide = FactoryBot.create(:edition, :welsh)
```

Comment:
Is there a reason we can't use `@welsh_edition` instead of creating a new one here?